File: CryptoTrader/login/chart.py
import datetime
from matplotlib import pyplot as plt
import requests
import pandas as pd
import mplfinance as mpf

#checks if crytpo exists

def availableCrypto():
    url = f'https://api.coingecko.com/api/v3/simple/price?ids=bitcoin%2Cethereum%2Cdogecoin%2Csolana&vs_currencies=gbp'
    response = requests.get(url)
    data = response.json()

    cryptoids = []

    for asset in data:
        cryptoids.append(asset)

    return cryptoids



#retrieve chart data for curreency

def getMarketCharts(coinId='bitcoin', vsCurrency='gbp', days='max', interval='daily'):
    cryptoIds = availableCrypto()

    if coinId in cryptoIds:
        url = f'https://api.coingecko.com/api/v3/coins/{coinId}/market_chart?vs_currency={vsCurrency}&days={days}&interval={interval}'
        payload = {"vsCurrency": vsCurrency, "days": days, "interval": interval}
        response = requests.get(url, params=payload)
        data = response.json()

        timestamp_list, price_list = [], []
        for price in data['prices']:
            timestamp_list.append(datetime.datetime.fromtimestamp(price[0]/1000))
            price_list.append(price[1])
        
        raw_data = {
            'timestamp' : timestamp_list,
            'price': price_list
        }
        df = pd.DataFrame(raw_data)
        return df
    else:
        print("Crypto selected is not available")
    
    return data

market_info = getMarketCharts('bitcoin', 'gbp', '30')

market_info.plot(y='price', x='timestamp', color='#4285F4')
plt.show()
print(market_info)

File: CryptoTrader/login/templates/static/chart.js
let myChart;

async function drawgraph(coin = "bitcoin") {
  const getMarketCharts = async (coinId, vsCurrency, days, interval) => {
    const url = `https://api.coingecko.com/api/v3/coins/${coinId}/market_chart?vs_currency=${vsCurrency}&days=${days}&interval=${interval}`;
    const response = await fetch(url);
    const data = await response.json();

    const timestamp_list = [];
    const price_list = [];

    const options = {
      weekday: "long",
      year: "numeric",
      month: "numeric",
      day: "numeric",
      hour: "numeric",
      minute: "numeric",
      second: "numeric",
    };

    data.prices.forEach((price) => {
      const date = new Date(price[0]);
      const dateTimeFormat = new Intl.DateTimeFormat("en-US", options);
      const parts = dateTimeFormat.formatToParts(date);
      let formattedDate = "";
      parts.forEach((part) => {
        if (part.type === "weekday") formattedDate += `${part.value}, `;
        if (part.type === "day") formattedDate += `${part.value}/`;
        if (part.type === "month") formattedDate += `${part.value}/`;
        if (part.type === "year") formattedDate += `${part.value}, `;
        if (part.type === "hour") formattedDate += `${part.value}:`;
        if (part.type === "minute") formattedDate += `${part.value}:`;
        if (part.type === "second") formattedDate += `${part.value}`;
      });
      timestamp_list.push(formattedDate);
      price_list.push(price[1]);
    });

    const raw_data = {
      labels: timestamp_list,
      datasets: [
        {
          label: `${coinId} price in ${vsCurrency}`,
          data: price_list,
          fill: false,
          borderColor: "rgb(66, 133, 244)",
          tension: 0.1,
        },
      ],
    };

    return raw_data;
  };
  const drawChart = async () => {
    const market_info = await getMarketCharts(coin, "gbp", "30", "daily");
    const ctx = document.getElementById("myChart").getContext("2d");

    if (myChart) {
      myChart.destroy();
    }

    myChart = new Chart(ctx, {
      type: "line",
      data: market_info,
    });
  };

  drawChart();
}
